irispy/sji.py

```diff
@@ -0,0 +1,158 @@
+import datetime
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+import sunpy.cm as cm
+import sunpy.instr.iris
+import sunpy.map
+import numpy as np
+import matplotlib.colors as colors
+from astropy.io import fits
+'''
+This module provides movie tools for level 2 IRIS SJI fits file
+'''
+__all__ = ["sji_fits_to_cube", "save_to_mp4", "dustbuster"]
+
+def sji_fits_to_cube(filelist, start=0, stop=None, skip=None, grid=False):
+    """
+    Read SJI files and return a MapCube. Inherits
+    sunpy.instr.iris.SJI_to_cube to stitch multiple sji fits
+    files stitched into one mapcube.
+    Also sets colormap and normalization
+    Assumes hdu index of 0.
+
+
+    Parameters
+    ----------
+    filelist: `str` or `list`
+        File to read, if single file string detected, will
+        create a list of length 1.
+
+    start: `int`
+        Temporal axis index to create MapCube from
+
+    stop: `int`
+        Temporal index to stop MapCube at
+
+    skip: `int`
+        Temporal index to skip over MapCube at
+
+
+    Returns
+    -------
+    iris_cube: `sunpy.map.MapCube`
+        A map cube of the SJI sequence
+    """
+
+    if type(filelist) == str:
+        filelist = [filelist]
+    iris_cube = sunpy.map.MapCube()
+    for fname in filelist[0:]:
+        newmap = sunpy.instr.iris.SJI_to_cube(fname)
```

Comment:
I think this could be merged like this and then we can refactor it later. I think what you could do is load all the files first and then create the one mapcube. Rather than appending the maps to an empty cube.